utils/build/docker/dotnet_http/parametric/Endpoints/ApmTestApiOtel.cs

```diff
@@ -0,0 +1,333 @@
+            }
+        }
+
+        // try extracting parent context from headers (remote parent)
+        if (requestBodyObject.TryGetValue("http_headers", out var headersList))
+        {
+            var extractedContext = _spanContextExtractor.Extract(
+                    ((Newtonsoft.Json.Linq.JArray)headersList).ToObject<string[][]>(),
+                    getter: GetHeaderValues!);
+            
+            _logger.LogInformation("Extracted SpanContext: {ExtractedContext}", extractedContext);
+
+            if (extractedContext is not null)
+            {
+                var parentTraceId = ActivityTraceId.CreateFromString(RawTraceId.GetValue(extractedContext) as string);
+                var parentSpanId = ActivitySpanId.CreateFromString(RawSpanId.GetValue(extractedContext) as string);
+                var flags = (SamplingPriority.GetValue(extractedContext) as int?) > 0 ? ActivityTraceFlags.Recorded : ActivityTraceFlags.None;
+
+                remoteParentContext = new ActivityContext(
+                    parentTraceId,
+                    parentSpanId,
+                    flags,
+                    AdditionalW3CTraceState.GetValue(extractedContext) as string,
+                    isRemote: true);
+            }
+        }
+
+        // sanity check that we didn't receive both a local and remote parent
+        if (localParentContext != null && remoteParentContext != null)
+        {
+            throw new ApplicationException(
+                "Both ParentId and HttpHeaders were provided to OtelStartSpan(). " +
+                "Provide one or the other, but not both.");
+        }
+
+        DateTimeOffset startTime = default;
+        if (requestBodyObject.TryGetValue("timestamp", out var timestamp))
+        {
+            startTime = new DateTime(1970, 1, 1) + TimeSpan.FromMicroseconds(Convert.ToInt64(timestamp));
+        }
+
+        var parentContext = localParentContext ?? remoteParentContext ?? default;
+
+        var kind = ActivityKind.Internal;
+
+        if (requestBodyObject.TryGetValue("span_kind", out var spanKind))
+        {
+            switch (Convert.ToInt64(spanKind))
+            {
+                case 1:
+                    kind = ActivityKind.Internal;
+                    break;
+                case 2:
+                    kind = ActivityKind.Server;
+                    break;
+                case 3:
+                    kind = ActivityKind.Client;
+                    break;
+                case 4:
+                    kind = ActivityKind.Producer;
+                    break;
+                case 5:
+                    kind = ActivityKind.Consumer;
+                    break;
+                default:
+                    kind = ActivityKind.Internal; // this is the default in Activity
+                    break;
+            }
+        }
+        
+        var activity = ApmTestApiActivitySource.StartActivity(
+            (string)requestBodyObject["name"],
```

Comment:
Super small nit: Although we always expect the incoming value to be a string, I suggest casting the object using the `as` operator. This will do the same thing as the hard `(string)` cast except if the object is not of type `string` then the `as` operator will return `null` whereas the cast would throw an `InvalidCastException`.

```suggestion
            requestBodyObject["name"] as string,
```